### ISSUE TITLE
RTG colors Amiberry vs. FS-UAE on Windows

### ISSUE THREAD

**Issue by andiweli** [midwan/amiberry#800]
**Describe the bug**
All my RTG modes on Workbench and games don't look like they have 16 mio. colora - they look more like 256 colors. Background pictures have visible gradients and so on (I had that issue also on 3.3).

**To Reproduce**
I select an RTG mode, for example 1280x720 with 16 or 32 Bit. I also set the WB pattern rendering to "best". 

**Expected behavior**
To see a clear colored picture as wallpaper.

**Screenshots**

FS-UAE for example
![image](https://user-images.githubusercontent.com/11447150/114007581-c3410500-9861-11eb-99ff-f6c480f1072c.jpeg)

Amiberry
![image](https://user-images.githubusercontent.com/11447150/114007656-d2c04e00-9861-11eb-947e-a117dcfee16e.jpeg)

**Desktop (please complete the following information):**

- OS: Raspbian Lite

Love Amiberry? Please consider supporting our collective:
👉  https://opencollective.com/amiberry/donate

**Comment by midwan** [comment]
I've tested this on my WB3.9 installation on a 64-bit distro, but I don't see the problem.
There is no color banding or dithering in my case, when using a 32-bit screen mode and a 24-bit JPEG as a backdrop image.
Do you have a 24-bit datatype installed?
Or perhaps this only happens on 32-bit installations for some reason (older SDL2?). I'll test it there a little later, need to prepare an environment first.

**Comment by andiweli** [comment]
Uh... good question. 
I downloaded a datatype from AmiNet recently. Never tested if it is 24bit... will try it later today!

**Comment by andiweli** [comment]
Do I have to set something special at "Color modes"?

The datatype is 32 bit. FS-UAE shows a clean picture when using the same HDF.

**Comment by midwan** [comment]
If you're using the same HDF, then I think it should work the same way (it's not a problem in AmigaOS at least).
Is the HDF large? Could you share a copy for testing reasons? Maybe we'll save some time troubleshooting if I can see the problem first-hand.

**Comment by andiweli** [comment]
> If you're using the same HDF, then I think it should work the same way (it's not a problem in AmigaOS at least).
> 
> Is the HDF large? Could you share a copy for testing reasons? Maybe we'll save some time troubleshooting if I can see the problem first-hand.

It is the same HDF I shared a few days ago with you...

I tried several datatypes now. Without success 😯

**Comment by midwan** [comment]
I tested this under a few different scenarios, they all worked normally for me:
- 64-bit, under Manjaro desktop
- 32-bit, RPI-OS Desktop, using SDL2
- 32-bit, RPI-OS Desktop, using Dispmanx

In all cases, using your HDF with the only modification being to select an RTG screenmode after the first boot, and increase the number of colors for that mode to 16.7M (from the 256 default).

You did change that slider all the way to the right, didn't you?

**Comment by andiweli** [comment]
Yes of course I did. 
Hmm then perhaps my Raspbian installation has some configuration error...?! 🙃

**Comment by midwan** [comment]
Or maybe this is a problem in KMSDRM only. I haven't been able to test that, yet.
It seems that KMSDRM has been getting a lot of bug fixes and updates in the SDL2 repo recently, so perhaps it's worth testing with the latest SDL2 version from source.
I imagine you compiled it yourself anyway, to get KMSDRM working on RPI OS Lite? The SDL2 version provided in the repositories does not have support for KMSDRM, so it wouldn't work otherwise.

**Comment by andiweli** [comment]
I managed to compile SDL2 now to 2.0.15 with all additional libs.
I used tis guide here: https://choccyhobnob.com/sdl2-2-0-8-on-raspberry-pi/

The colors still aren't working for me :(

**Funny note...**
I have a AmigaOS 3.9 HDF too - the colors there are too looking like 256 colors - even 32 Bit is selected.
But if I select 16 Bit, everything looks fine.
Switching back to 32 Bit again has this 256 colors look.

I tried it on my 3.1.4.1 installation too, but even with 16 Bit all Picasso modes give me that 256 color look.
